src/libopensc/card-authentic.c

```diff
@@ -917,37 +873,35 @@ authentic_write_binary(struct sc_card *card, unsigned int idx,
 	       "offs:%i,count:%"SC_FORMAT_LEN_SIZE_T"u,max_send_size:%"SC_FORMAT_LEN_SIZE_T"u",
 	       idx, count, card->max_send_size);
 
-	/* see comments for authentic_read_binary() */
-	sc_log(ctx, "reader flags 0x%lX", card->reader->flags);
-	if (count > 255 && !(card->reader->flags & SC_READER_HAS_WAITING_AREA))
-		LOG_TEST_RET(ctx, SC_ERROR_INVALID_DATA, "Invalid size of the data to read");
+	apdu = calloc(1, sizeof(struct sc_apdu));
+	if(!apdu)
+		LOG_TEST_RET(ctx, SC_ERROR_OUT_OF_MEMORY, "cannot allocate APDU");
```

Comment:
same here, if you use a local variable, you're saving the hassle of cleaning up the memory.

Reply:
Also done.